xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/resetpasswordinline.vm

```diff
@@ -0,0 +1,293 @@
+    <div class="panel-title">$services.localization.render('xe.admin.passwordReset.title')</div>
+  </div>
+  <div class="panel-body">
+#end
+
+## The name of the class used for storing password reset verification data.
+#set ($verifClass = 'XWiki.ResetPasswordRequestClass')
+#set ($userName = "$!request.get('u')")
+#set ($verifCode = "$!request.get('v')")
+#if ($userName == '' && $verifCode == '')## First step, display the form requesting the username
+#resetPasswordBoxStart("default")
+$services.localization.render('xe.admin.passwordReset.instructions')
+
+<form method="post" action="$services.security.authentication.getAuthenticationURL('reset', $NULL)" class="xformInline" id="resetPasswordForm">
+    <div>
+        <input type="hidden" name="form_token" value="$!{services.csrf.getToken()}" />
+        <label for="u">$services.localization.render('xe.admin.passwordReset.username.label')</label>
+        <input type="text" id="u" name="u"/>
+        <span class="buttonwrapper">
+            <input type="submit" value="$services.localization.render('xe.admin.passwordReset.submit')" class="button"/>
+        </span>
+    </div>
+</form>
+#elseif ($userName != '' && $verifCode == '')
+    ## Second step, generate the verification string, store it, and send the email
+    ## TODO: Once the usernames are not bound to the XWiki space, revisit this code
+    #if ($userName.indexOf('.') != -1)
+        #set ($userDoc = $xwiki.getDocumentAsAuthor(${userName}))
+    #else
+        #set ($userDoc = $xwiki.getDocumentAsAuthor("XWiki.${userName}"))
+    #end
+    #set ($userExists = $services.user.exists($userName))
+    #if (!$userExists)
+        #resetPasswordBoxStart("warning")
+        $services.localization.render('xe.admin.passwordReset.error.noUser', ["<strong>${escapetool.xml($userName)}</strong>"])
+    #elseif ($userDoc.getObject('XWiki.LDAPProfileClass'))
+        #resetPasswordBoxStart("warning")
+        $services.localization.render('xe.admin.passwordReset.error.ldapUser', ["<strong>${escapetool.xml($userName)}</strong>"])
+    #else
+        #set ($userProperties = $services.user.getProperties($userName))
+        #set ($userEmail = $userProperties.email)
+        #if ("$!userEmail" == '')
+            #resetPasswordBoxStart("error")
+            $services.localization.render('xe.admin.passwordReset.error.noEmail')</div>
+        #else
+            ## Find the object that will hold the verification string
+            #set ($verifObj = '')
+            #set ($verifObj = $userDoc.getObject($verifClass, true))
+            ## Generate a random string
+            #set ($verifStr = $util.generateRandomString(30))
+            ## If the class is correctly configured, the string should automatically be stored as a hash
+            #set ($discard = $verifObj.set('verification', $verifStr))
+            #set ($discard = $userDoc.saveAsAuthor($services.localization.render('xe.admin.passwordReset.versionComment'), true))
```

Comment:
Using save() instead of saveAsAuthor() would also fix https://jira.xwiki.org/browse/XWIKI-16661.